src/libs/SidebarUtils.ts

```diff
@@ -278,6 +297,21 @@ function getOptionData({
     result.shouldShowSubscript = ReportUtils.shouldReportShowSubscript(report);
     result.pendingAction = report.pendingFields?.addWorkspaceRoom ?? report.pendingFields?.createChat;
     result.brickRoadIndicator = hasErrors || hasViolations ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : '';
+    const oneTransactionThreadReportID = ReportActionsUtils.getOneTransactionThreadReportID(report.reportID, ReportActionsUtils.getAllReportActions(report.reportID));
```

Comment:
Instead of putting this logic here, how about placing it with the parent report `OptionRowLHNData`? I am suggesting this to keep the `shouldDisplayViolations` in a single place.

https://github.com/Expensify/App/blob/dd96852320612fd5188c95e659fce3814150af82/src/components/LHNOptionsList/OptionRowLHNData.tsx#L37-L39

```js
const oneTransactionThreadReportID = ReportActionsUtils.getOneTransactionThreadReportID(reportID, ReportActionsUtils.getAllReportActions(reportID));
let oneTransactionThreadViolation = false;
if (oneTransactionThreadReportID) {
  const oneTransactionThreadReport = getReport(oneTransactionThreadReportID);

  if (
      ReportUtils.shouldDisplayTransactionThreadViolations(
          oneTransactionThreadReport,
          transactionViolations,
          ReportActionsUtils.getAllReportActions(reportID)[oneTransactionThreadReport?.parentReportActionID ?? '-1'],
      )
  ) {
    oneTransactionThreadViolation = true;
  }
}

const shouldDisplayViolations = canUseViolations && (ReportUtils.shouldDisplayTransactionThreadViolations(fullReport, transactionViolations, parentReportAction) || oneTransactionThreadViolation);
```

Reply:
I think we should let the ```SizebarUtils``` handle the logic to display the RBR in case one transaction report has violations because:
1. We have already handled the logic to pin the one transaction report if it has a violation in ```SizebarUtils```. So the logic to display RBR should be placed here for easier management.

2. Reduce the complexity of the ```OptionRowLHNData``` file.

cc @robertjchen What do you think?